src/uv-common.c

```diff
@@ -100,6 +100,17 @@ void* uv__realloc(void* ptr, size_t size) {
   return NULL;
 }
 
+void* uv__reallocf(void* ptr, size_t size) {
+  void* newptr;
+
+  newptr = uv__realloc(ptr, size);
+  if (newptr == NULL)
+    if (size > 0)
+      uv__free(ptr);
```

Comment:
Maybe wrapping these conditionals into one?

Reply:
You mean with `&&`? The two-if-statements style is pretty pervasive throughout libuv.

Reply:
Yeah, I meant that. It was just a suggestion. I'm fine either way.